astropy/uncertainty/core.py

```diff
@@ -6,9 +6,21 @@
 import builtins
 
 import numpy as np
+from numpy.lib.stride_tricks import DummyArray
 
 from astropy import stats
 from astropy import units as u
+from astropy.utils.compat.numpycompat import NUMPY_LT_1_23, NUMPY_LT_2_0
+
+if NUMPY_LT_2_0:
+    from numpy.core.multiarray import normalize_axis_index
+    from numpy.lib.function_base import _parse_gufunc_signature
```

Comment:
I just looked and `_parse_gufunc_signature` hasn't changed in 7 years, so this is probably not an issue, but... should we be worried it might change as a private function?

Reply:
We should, and really the "true" gufunc parsing in the ufunc machinery should be exposed, which I hope to do one day, but for now I think it is fine to rely on our testing. Note that this is used in `Masking` too, where tests are a little more rigorous.